### PR TITLE
feat(shared): adapt useGuidelineDesignTokens to use main and system font

### PR DESCRIPTION
**Pull request by celine-s**
ticket: https://app.clickup.com/t/86775v47c

### CODE REVIEW

File: packages/shared/src/helpers/mapToGuidelineDesignTokens.ts [hunk 1]
```diff
@@ -41,11 +41,13 @@ const transformObjectValues = (key: string, cssStyles: TokenValues, value: Direc
     }
 };
 
-const transformStringValues = (key: string, cssStyles: TokenValues, value: string) => {
+const transformStringValues = (key: string, cssStyles: TokenValues, value: string, mainFontFamily: string) => {
     cssStyles.hover = cssStyles.hover || {};
     switch (key) {
         case DesignTokenPropertiesEnum.family:
-            cssStyles.fontFamily = value;
+            const useMainFont = value === 'inherit' ? mainFontFamily : undefined;
```
Comment (triggertoo) [inline packages/shared/src/helpers/mapToGuidelineDesignTokens.ts:48]:
```suggestion
            const mainFont = value === 'inherit' ? mainFontFamily : undefined;
```

File: packages/shared/src/helpers/mapToGuidelineDesignTokens.ts [hunk 2]
```diff
@@ -41,11 +41,13 @@ const transformObjectValues = (key: string, cssStyles: TokenValues, value: Direc
...
+            const newValue = value === 'default' ? 'system-ui' : value;
```
Comment (triggertoo) [inline packages/shared/src/helpers/mapToGuidelineDesignTokens.ts:49]:
```suggestion
            const fallbackFont = value === 'default' ? 'system-ui' : value;
```

File: packages/shared/src/helpers/mapToGuidelineDesignTokens.ts [hunk 3]
```diff
@@ -119,10 +121,14 @@ const transformStringValues = (key: string, cssStyles: TokenValues, value: strin
 export const mapToGuidelineDesignTokens = (dataToTransform: Partial<Record<string, DesignTokenProperties>>) => {
     const transformedDesignTokens: TransformedDesignTokens = {};
     const enrichedDataToTransform = provideDefaultCalloutColors(dataToTransform);
+    const mainFontFamily =
+        (dataToTransform['main_font'] as Record<DesignTokenPropertiesEnum, string>)?.family === 'default'
```
Comment (triggertoo) [inline packages/shared/src/helpers/mapToGuidelineDesignTokens.ts:125]: couldn't the above logic be reused here?

File: packages/shared/src/helpers/mapToGuidelineDesignTokens.ts (hunk 2 above)
Comment (celine-s) [reply packages/shared/src/helpers/mapToGuidelineDesignTokens.ts]: thanks! 👍

File: packages/shared/src/helpers/mapToGuidelineDesignTokens.ts (hunk 1 above)
Comment (celine-s) [reply packages/shared/src/helpers/mapToGuidelineDesignTokens.ts]: 👍

File: packages/shared/src/helpers/mapToGuidelineDesignTokens.ts (hunk 3 above)
Comment (celine-s) [reply packages/shared/src/helpers/mapToGuidelineDesignTokens.ts]: refactored & tests added

**Review by Walkeezy — APPROVED** [review]

**Review by triggertoo — APPROVED** [review]

**Pull request merged**